Source/WebInspectorUI/UserInterface/Views/InlineSwatch.js

```diff
@@ -362,28 +400,24 @@ WI.InlineSwatch = class InlineSwatch extends WI.Object
         this.dispatchEventToListeners(WI.InlineSwatch.Event.Activated);
 
         switch (this._type) {
-        case WI.InlineSwatch.Type.CubicBezierTimingFunction:
-            this._valueEditor.cubicBezierTimingFunction = value;
-            break;
-
-        case WI.InlineSwatch.Type.BoxShadow:
-            this._valueEditor.boxShadow = value;
-            break;
-
         case WI.InlineSwatch.Type.Color:
-            this._valueEditor.color = value;
             this._valueEditor.focus();
             break;
 
         case WI.InlineSwatch.Type.Gradient:
             this._valueEditor.gradient = value;
             break;
 
-        case WI.InlineSwatch.Type.SpringTimingFunction:
-            this._valueEditor.springTimingFunction = value;
-            break;
 
         case WI.InlineSwatch.Type.Variable: {
+            if (this._variableType) {
+                let colorVariable = this._delegate?.inlineSwatchColorVariableCompute(this, value.toString());
+                if (colorVariable) {
+                    this._valueEditor.color = WI.Color.fromString(colorVariable.value);
+                    this._valueEditor.focus();
```

Comment:
wait if `this._type` is `WI.InlineSwatch.Type.Variable` then the `_valueEditor` will be an object containing a `CodeMirror` instance, not a `WI.ColorPicker`, right?  what is the point of this line of code?

Reply:
it's needed because if you note above within this switch case, I still use this.type and I did this cause I felt that this was a scenario where a color variable would have to do a lot of different processes as opposed to a normal color inline swatch so, I used this._type but within the variable type I took case of the case of the color variable